### PR TITLE
expose refresh token retrieving mechanism in profile

### PR DESCRIPTION
**Pull request by skkestrel**
This PR facilitates the future addition of "az acr login" command. It is necessary for the our authorization server to read the user's refresh token in order to obtain access tokens to authorize the user against ARM RBAC.

This PR adds the functions:
- get_refresh_credentials (in Profile)
- retrieve_token_entry_for_user (in CredsCache)
- retrieve_cred_for_service_principal (in CredsCache)

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -38,6 +38,7 @@
 _STATE = 'state'
 _USER_TYPE = 'type'
 _USER = 'user'
+_USER_AUTHORITY = '_authority'
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/_profile.py]: This doesn't appear being used.

**Comment by skkestrel** [comment]
Unfortunately the addition of the "acr login" command is blocked until we figure out some more things on our end regarding the deployment of the authorization server, etc. The CLI code does exist but there are no proper endpoints defined so the command would not be usable anyways. If we merge this PR then it would streamline the process when we are ready to add "acr login". What do you think?

**Comment by skkestrel** [comment]
Unit tests added:
- test_get_refresh_credentials_for_spn
- test_get_refresh_credentials_for_graph_client
- test_get_refresh_credentials

and these were modified to additionally check for refresh tokens/credentials:
- test_credscache_new_token_added_by_adal
- test_credscache_load_tokens_and_sp_creds
- test_credscache_add_new_sp_creds

**Comment by yugangw-msft** [comment]
>If we merge this PR then it would streamline the process when we are ready to add "acr login". 

There are really nothing more to streamline, if you include this change with the new commands, which also act as good tests. No existing commands need this, correct? So merging it right now doesn't help instead it introduces risk to the upcoming release.

**Comment by skkestrel** [comment]
OK. Will recreate PR when acr login is ready to ship.

**Comment by yugangw-msft** [comment]
@skkestrel, sounds good.

**Pull request closed without merging**